sympy/tensor/riemannian_geometry.py

```diff
@@ -0,0 +1,1159 @@
+# -*- coding: utf-8 -*-
+
+from sympy.matrices import Matrix
+from sympy.core import Add, diff, Symbol
+from sympy.simplify import simplify
+from sympy.tensor.arraypy import Arraypy, Tensor, matrix2arraypy, \
+    matrix2tensor, list2arraypy, list2tensor
+
+"""Module riemannian_geometry contains functions for working with tensor fields:
+- the calculation of the scalar product;
+- the Christoffel symbols of the first and second kind;
+- the covariant derivative of the curvature tensor;
+- the Ricci tensor;
+- scalar and sectional curvature.
+
+To implement the functions used modules: matrices and tensor
+(with classes arraypy and tensor). All functions take arguments,
+the types of which may be such as list, matrix, or array arraypy tensor.
+Some functions have optional parameter indicating the type of the function result.
+Starting index of arguments with type Arraypy or Tensor is not necessarily
+and by default equal to 0. The function determines the range of the index
+in array to return the object with the same range of index.
+
+Functions are work with multidimensional arrays arraypy and tensors,
+classes and methods are contained in the module arraypy.
+
+"""
+
+
+def scal_prod(X, Y, g):
+    """Returns scalar product of vectors g(X,Y).
+
+    Examples:
+    =========
+
+    >>> from sympy.tensor.riemannian_geometry import scal_prod
+    >>> from sympy import symbols, cos
+    >>> from sympy.tensor.arraypy import Arraypy, Tensor
+    >>> x1, x2 = symbols('x1, x2')
+
+    X, Y is a vector or a vector field. They can be a list,
+    one-dimensional arraypy or Tensor with valence of indices (+1):
+
+    >>> X = [1, 2]
+    >>> Y = [3, 4]
+
+    g is a metric tensor must be symmetric matrix, arraypy or tensor
+    with valence of indices (-1, -1):
```

Comment:
Why does it have to have valence (-1, -1) ? I think it can be converted to that valence, can't it?